cli/scripting_test.go

```diff
@@ -0,0 +1,244 @@
+					input, err := json.Marshal(ScriptingRunInput{
+						ID:   id,
+						Args: []string{tmpFile},
+					})
+					require.NoError(t, err)
+					resp := &OutcomeResponse{}
+					require.NoError(t, execCLICommandInputOutput(t, c, scriptingRun(), input, resp))
+					assert.True(t, resp.Successful())
+				},
+				"RunWithEmptyIDFails": func(ctx context.Context, t *testing.T, c *cli.Context, tmpDir string) {
+					input, err := json.Marshal(ScriptingRunInput{Args: []string{"./"}})
+					require.NoError(t, err)
+					resp := &OutcomeResponse{}
+					assert.Error(t, execCLICommandInputOutput(t, c, scriptingRun(), input, resp))
+					assert.False(t, resp.Successful())
+				},
+				"RunWithNonexistentIDFails": func(ctx context.Context, t *testing.T, c *cli.Context, tmpDir string) {
+					input, err := json.Marshal(ScriptingRunInput{
+						ID:   "foo",
+						Args: []string{"./"},
+					})
+					require.NoError(t, err)
+					resp := &OutcomeResponse{}
+					require.NoError(t, execCLICommandInputOutput(t, c, scriptingRun(), input, resp))
+					assert.False(t, resp.Successful())
+				},
+				"RunScriptSucceeds": func(ctx context.Context, t *testing.T, c *cli.Context, tmpDir string) {
+					id := createScriptingFromCLI(t, c, testutil.ValidGolangScriptingHarnessOptions(tmpDir))
+					setupScriptingFromCLI(t, c, id)
+					input, err := json.Marshal(ScriptingRunScriptInput{
+						ID:     id,
+						Script: testutil.GolangMainSuccess(),
+					})
+					require.NoError(t, err)
+					resp := &OutcomeResponse{}
+					require.NoError(t, execCLICommandInputOutput(t, c, scriptingRunScript(), input, resp))
+					assert.True(t, resp.Successful())
+				},
+				"RunScriptWithEmptyIDFails": func(ctx context.Context, t *testing.T, c *cli.Context, tmpDir string) {
+					input, err := json.Marshal(ScriptingRunScriptInput{Script: testutil.GolangMainSuccess()})
+					require.NoError(t, err)
+					resp := &OutcomeResponse{}
+					assert.Error(t, execCLICommandInputOutput(t, c, scriptingRunScript(), input, resp))
+					assert.False(t, resp.Successful())
+				},
+				"RunScriptWithNonexistentIDFails": func(ctx context.Context, t *testing.T, c *cli.Context, tmpDir string) {
+					input, err := json.Marshal(ScriptingRunScriptInput{
+						ID:     "foo",
+						Script: testutil.GolangMainSuccess(),
+					})
+					require.NoError(t, err)
+					resp := &OutcomeResponse{}
+					require.NoError(t, execCLICommandInputOutput(t, c, scriptingRunScript(), input, resp))
+					assert.False(t, resp.Successful())
+				},
+				"BuildSucceeds": func(ctx context.Context, t *testing.T, c *cli.Context, tmpDir string) {
+					id := createScriptingFromCLI(t, c, testutil.ValidGolangScriptingHarnessOptions(tmpDir))
+					setupScriptingFromCLI(t, c, id)
+					tmpFile := filepath.Join(tmpDir, "main.go")
+					require.NoError(t, ioutil.WriteFile(tmpFile, []byte(`package main; import "os"; func main() { os.Exit(0) }`), 0755))
```

Comment:
Why not use `testutil.GolangMainSuccess()`?

Reply:
That was an oversight. Fixed it.